jenkins/Jenkinsfile

```
@@ -1,374 +1,496 @@
+                                                --customization-script ../prepare-custom-leonardo-jupyter-dataproc-image.sh \
+                                                --zone $ZONE \
+                                                --gcs-bucket $DATAPROC_IMAGE_BUCKET \
+                                                --project-id=$GOOGLE_PROJECT \
+                                                --disk-size=50
+                                        """
+                                    )
+                                }
+                            }
+                        }
+                }
+
+                stage('Run GCE image generation script') {
+                    when { expression { GENERATE_GCE_IMAGE == "true" } }
+                        steps {
+                            sh """
+                                docker pull gcr.io/compute-image-tools/daisy:latest
+                            """
+
+                            script {
+                                String daisyImage = "gcr.io/compute-image-tools/daisy:latest"
+                                sh """
+                                    # Create the Daisy scratch bucket if it doesn't exist. The Daisy workflow will clean it up at the end.
+                                    gsutil ls $GCE_IMAGE_BUCKET || gsutil mb -p $GOOGLE_PROJECT -l $REGION $GCE_IMAGE_BUCKET
+
+                                    docker pull $daisyImage
+                                    docker run -i --rm -v ${WORKSPACE}/jenkins/gce-custom-images:/daisy_source_files \
+                                        $daisyImage \
+                                        -project $GOOGLE_PROJECT \
+                                        -zone $ZONE \
+                                        -oauth /daisy_source_files/$GOOGLE_CREDENTIAL_FILE_NAME \
+                                        -gcs_path $GCE_IMAGE_BUCKET \
+                                        -default_timeout 45m \
```

Wonder if 45m is too short -- pulling all our docker images takes time.

---

It's hovered around 20 mins for me. Another reason I didn't want to make it longer is because Daisy keeps waiting until the timeout even if something in the Daisy workflow fails early so I didn't want the Jenkins job to take longer than necessary in cases of failures.